core/src/main/java/com/scalar/db/common/CommonDistributedStorageAdmin.java

```diff
@@ -498,6 +501,188 @@ public StorageInfo getStorageInfo(String namespace) throws ExecutionException {
     }
   }
 
+  @Override
+  public void createVirtualTable(
+      String namespace,
+      String table,
+      String leftSourceNamespace,
+      String leftSourceTable,
+      String rightSourceNamespace,
+      String rightSourceTable,
+      VirtualTableJoinType joinType,
+      Map<String, String> options)
+      throws ExecutionException {
+    StorageInfo storageInfo = getStorageInfo(leftSourceNamespace);
+    switch (storageInfo.getMutationAtomicityUnit()) {
+      case STORAGE:
+        break;
```

Comment:
I think it's possible that the left namespace's atomicity unit is STORAGE and the right one is non-STORAGE when using the multi-storage feature. We don't need to prohibit this combination?

Reply:
`CommonDistributedStorageAdmin` is for each adapter, so it doesn’t handle that validation.
For multi-storage scenarios, the validation is performed in the following part:

https://github.com/scalar-labs/scalardb/pull/3180/files#diff-89fa12120bc8db2859aec83840628c00c55ef739084aa207c6d78bdb17c20f8fR328-R345

Thanks!